Add PHP 8.0 Support

stringerbell opened the pull request:
<!--
Fill in the relevant information below to help triage your issue.

Pick the target branch based on the following criteria:
  * Documentation improvement: master branch
  * Bugfix: master branch
  * QA improvement (additional tests, CS fixes, etc.) that does not change code
    behavior: master branch
  * New feature, or refactor of existing code: develop branch

You MUST provide a signoff in your commits for us to be able to accept your
patch; you can do this by providing either the --signoff or -s flag when using
"git commit". Please see the project contributing guide and
https://developercertificate.org for details.
-->

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | yes
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

Fixes #8 

I noticed this makes  `composer cs-check` and `composer cs-fix` break, which I'd imagine is a blocker for this. I wasn't sure if it was acceptable to include those changes as well in this. 

The error is:
```
Fatal error: Array and string offset access syntax with curly braces is no longer supported in /path/to/laminas-development-mode/vendor/squizlabs/php_codesniffer/CodeSniffer/CLI.php on line 453
```

I'm happy to make a separate PR, or include upgrading `laminas/laminas-coding-standard` in here if that's desirable.

Edit: Here's a diff of just the phpcs changes:
https://github.com/laminas/laminas-development-mode/compare/3.3.x...stringerbell:update-laminas-coding-standard

Adding visibility modifiers to the constants would break 5.6 anyway, so perhaps it ought to all get mashed in here?

<!--
Tell us about why this change is necessary:
- Are you fixing a bug or providing a failing unit test to demonstrate a bug?
  - How do you reproduce it?
  - What did you expect to happen?
  - What actually happened?
  - TARGET THE master BRANCH

- Are you adding documentation?
  - TARGET THE master BRANCH

- Are you providing a QA improvement (additional tests, CS fixes, etc.) that
  does not change behavior?
  - Explain why the changes are necessary
  - TARGET THE master BRANCH

- Are you fixing a BC Break?
  - How do you reproduce it?
  - What was the previous behavior?
  - What is the current behavior?
  - TARGET THE master BRANCH

- Are you adding something the library currently does not support?
  - Why should it be added?
  - What will it enable?
  - How will the code be used?
  - TARGET THE develop BRANCH

- Are you refactoring code?
  - Why do you feel the refactor is necessary?
  - What types of refactoring are you doing?
  - TARGET THE develop BRANCH
-->

stringerbell committed:
Change composer dependencies

- Modify composer.json to provide support for PHP 8.0 by adding the
constraint ~8.0.0
- Modify composer.json to drop support for PHP less than 7.3
- Modify composer.json to implement phpunit 9.3 which supports PHP 7.3+

Signed-off-by: Dann Stockton <dannstockton@gmail.com>

stringerbell committed:
Modify travis.yml

- Modify .travis.yml to ignore platform requirements when installing
composer dependencies (simply add --ignore-platform-reqs to
COMPOSER_ARGS env variable)

- Modify .travis.yml to add PHP 8.0 to the matrix (NOTE: Do not allow
failures as PHP 8.0 has a feature freeze since 2020-08-04!)

Signed-off-by: Dann Stockton <dannstockton@gmail.com>

stringerbell committed:
Ignore .phpunit.result.cache

Signed-off-by: Dann Stockton <dannstockton@gmail.com>

stringerbell committed:
Add return types

Signed-off-by: Dann Stockton <dannstockton@gmail.com>

stringerbell committed:
Change calls to assertStringContainsString

Signed-off-by: Dann Stockton <dannstockton@gmail.com>

stringerbell committed:
Run phpunit --migrate-configuration

Signed-off-by: Dann Stockton <dannstockton@gmail.com>

boesing commented on .travis.yml:
```diff
@@ -12,30 +12,22 @@ env:
       env:
         - DEPS=latest
-    - php: 7.1
+    - php: nightly
```
Could you change the travis configuration as in laminas/laminas-servicemanager#64?
Travis added PHP 8.0 [yesterday](https://travis-ci.community/t/php-8-0-missing/10132/23) support for the matrix and thus we can add `8.0` aswell as `nightly` where nightly is allowed to fail.

There is an example in the linked PR 👍

stringerbell replied:
Updated. I wasn't quite sure about adding the env vars here, but I'm happy to do that if that's desired: https://github.com/laminas/laminas-servicemanager/blob/cd5686e97ca9637ee69244a7b2396bbb5113ec4e/.travis.yml#L19-L22

Also, any thoughts about including cs-check/cs-fix changes in this request too?

froschdesign replied:
@stringerbell
> Also, any thoughts about including cs-check/cs-fix changes in this request too?

Please create a separate pull request for this because the topic of this pull request and the related issue report is the support of PHP 8 and this is enough. It is clearer for the reviewers and makes it easier for you too.

Thanks in advance! 👍

stringerbell committed:
Update to include PHP 8.0 and nightly

- Allow nightly to fail

Signed-off-by: Dann Stockton <dannstockton@gmail.com>

weierophinney committed:
qa: revert formatting and schema location changes from phpunit.xml.dist

When you migrate the dist file, it sets the schema location to the canonical URL for it; this then adds latency when running PHPUnit, as it needs to fetch it before it can validate the configuration.
We can use the locally installed version still; we just need to update the structure to follow the schema.

Additionally, we tend to favor 4-space indentiation everywhere, so I reverted indentation changes.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
Adds CHANGELOG entry for #9

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

stringerbell commented:
Thanks all for your help, and thanks for your fixups, @weierophinney! 🙌🏻
